app/middleware/simplified-account/enforce-cannot-do-admin-action-on-self.middleware.js

```diff
@@ -0,0 +1,11 @@
+const { NotFoundError } = require('../../errors')
+
+module.exports = function getAdminActionOnSelfMiddleware (errorMessage) {
+  return function adminActionOnSelf (req, res, next) {
+    if (req.params.externalUserId === req.user.externalId) {
+      next(new NotFoundError(errorMessage))
```

Comment:
I think a `403` or `409` makes more sense, if we have errors for either of those

Reply:
yes, change to NotAuthorisedError

Reply:
hmm actually I don't think that's appropriate either

Reply:
To allow a specific error message to be displayed to the user, I've removed the middleware and inserted a conditional in the controller. I've applied the same to the remove user controller and added tests for this.